src/main/scala/cromwell/binding/command/ParameterCommandPart.scala

```diff
@@ -3,17 +3,19 @@ package cromwell.binding.command
 import cromwell.binding.WdlValue
 import cromwell.binding.types.WdlType
 
+import scala.util.Try
+
 case class ParameterCommandPart(wdlType: WdlType, name: String) extends CommandPart {
   override def toString: String = "${" + s"$wdlType $name" + "}"
 
-  def instantiate(parameters: Map[String, WdlValue]): String = {
-    val paramValue = parameters.getOrElse(name, {
-      throw new UnsupportedOperationException(s"Parameter $name not found")
-    })
-    if (wdlType != paramValue.wdlType) {
-      throw new UnsupportedOperationException(s"Incompatible type for $name: need a $wdlType, got a ${paramValue.wdlType}")
-    }
-    /* TODO: this also is a gross over-simplification */
-    paramValue.value.toString
-  }
+  def instantiate(parameters: Map[String, WdlValue]): Try[String] =
```

Comment:
Instantiating a command line is the process of taking a command in this form:

```
sh script.sh ${var1} -o ${var2}
```

This command is stored as a `Seq[CommandPart]` in the Command class (e.g. [`sh script.sh `, `${var1}`, ` -o `, `${var2}`]).  Then, given a map of variable -> value:

```
{
  "var1": "foo",
  "var2": "bar"
}
```

It calls `instantiate()` on each part, and passes this map.  The ParameterCommandPart are the `${var1} and `${var2}` pieces and they lookup `var1` and `var2` in that map.

The command that's returned from `Command.instantiate()` is:

```
sh script.sh foo -o bar
```